### PR TITLE
fix Charlist <-> String conversion to use unicode

### LINKED ISSUE #25: charlist to string and UTF8
I think <https://github.com/gleam-lang/erlang/blob/main/src/gleam/erlang/charlist.gleam> should use

```
unicode:characters_to_binary(CharList, utf8, utf8)
```

instead of

```
erlang:list_to_binary(CharList)
```

etc.

For example:

```
Eshell V13.1.2  (abort with ^G)
1> erlang:list_to_binary("Jérôme").
<<"Jérôme">>
2> unicode:characters_to_binary("Jérôme", utf8, utf8).
<<"Jérôme"/utf8>>
```

Got caught with my first try with Gleam when I wanted to:

```
gleam run Jérôme
```

with

```
import gleam/io
import gleam/string
import gleam/erlang

pub fn main() {
  let args = erlang.start_arguments()
  let name = string.join(args, " ")
  io.println("Hello " <> name <> "!")
}
```

`string.join` fails.

### PR DESCRIPTION
**Pull request by jeromede**
Fix #25

### CODE REVIEW

File: src/gleam/erlang/charlist.gleam [hunk 1]
```diff
@@ -10,9 +10,11 @@
 pub external type Charlist
 
 /// Transform a charlist to a string
+/// Note: Erlang default encoding is utf8, so we're good
```
[the remark this answers is not in the document]
Comment (jeromede) [reply src/gleam/erlang/charlist.gleam]: OK, yes, just understood the comment conventions. See new commit.

File: test/gleam/erlang/charlist_test.gleam [hunk 2]
```diff
@@ -8,6 +8,16 @@ pub fn to_string_test() {
...
 }
 
+external fn lists_duplicate(n: Int, elem: Int) -> charlist.Charlist =
+  "lists" "duplicate"
+
+pub fn from_char_list_test() {
+  // Test conversion from a char list generated by Erlang
+  assert "ééé" =
+    lists_duplicate(3, 0xE9)
+    |> charlist.to_string()
```
[the remark this answers is not in the document]
Comment (jeromede) [reply test/gleam/erlang/charlist_test.gleam:18]: Héhéhé

File: src/gleam/erlang/charlist.gleam (hunk 1 above)
[the remark this answers is not in the document]
Comment (lpil) [reply src/gleam/erlang/charlist.gleam]: Thank you!

**Comment by jeromede** [comment]
Sorry :-/ Done

**Comment by jeromede** [comment]
At some point, I’ll get the comment conventions...

**Review by lpil — COMMENTED** [review]
CI is busted! I'll fix that

**Review by lpil — APPROVED** [review]

**Pull request merged**